app/services/EssttpBarsService.scala

```diff
@@ -63,16 +63,21 @@ class EssttpBarsService @Inject() (
       )
       .flatMap { result =>
         auditService.auditBarsCheck(journey, bankDetails, typeOfBankAccount, result)
-        barsVerifyStatusConnector
-          .update(taxId)
-          .map { verifyStatus =>
-            // here we catch a lockout BarsStatus condition,
-            // and force a TooManyAttempts (BarsError) response
-            verifyStatus.lockoutExpiryDateTime
-              .fold(result) { expiry =>
-                result.flatMap(resp => Left(TooManyAttempts(resp, expiry)))
+        result match {
+          case Left(_: BarsValidateError) | Left(_: SortCodeOnDenyListValidateError) =>
```

Comment:
is there a reason that `SortCodeOnDenyListValidateError` isn't a `BarsValidateError`?

Reply:
the main reason is this 
```
sealed trait BarsValidateError extends BarsError {
  override val barsResponse: ValidateResponse
}

sealed trait SortCodeOnDenyListValidateError extends BarsError {
  override val barsResponse: SortCodeOnDenyList
}
```
So we can differentiate easily between different types of error response and because a ValidateResponse is a different shape than a SortCodeOnDenyList

Reply:
I will try this 
```sealed trait BarsError {
  val barsResponse: BarsResponse
}
sealed trait BarsValidateError extends BarsError {
  override val barsResponse: BarsResponse
}
sealed trait SortCodeOnDenyListValidateError extends BarsValidateError {
  override val barsResponse: BarsResponse
}
sealed trait BarsVerifyError extends BarsError {
  override val barsResponse: VerifyResponse
}
```

Reply:
updated